sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -0,0 +1,187 @@
+            :end-before: [END create_eventhub_consumer_client_async]
+            :language: python
+            :dedent: 4
+            :caption: Create a new instance of the EventHubConsumerClient.
+    """
+
+    def __init__(self, host, event_hub_path, credential, **kwargs):
+        # type:(str, str, Union[EventHubSharedKeyCredential, EventHubSASTokenCredential, TokenCredential], Any) -> None
+        """
+        :param host: The hostname of the Event Hub.
+        :type host: str
+        :param event_hub_path: The path of the specific Event Hub to connect the client to.
+        :type event_hub_path: str
+        :param credential: The credential object used for authentication which implements particular interface
+         of getting tokens. It accepts ~azure.eventhub.EventHubSharedKeyCredential,
+         ~azure.eventhub.EventHubSASTokenCredential, credential objects generated by the azure-identity library and
+         objects that implement get_token(self, *scopes) method.
+        :keyword network_tracing: Whether to output network trace logs to the logger. Default
+         is `False`.
+        : network_tracing: bool
+        :param http_proxy: HTTP proxy settings. This must be a dictionary with the following
+         keys: 'proxy_hostname' (str value) and 'proxy_port' (int value).
+         Additionally the following keys may also be present: 'username', 'password'.
+        :type http_proxy: dict[str, Any]
+        :param auth_timeout: The time in seconds to wait for a token to be authorized by the service.
+         The default value is 60 seconds. If set to 0, no timeout will be enforced from the client.
+        :type auth_timeout: float
+        :param user_agent: The user agent that needs to be appended to the built in user agent string.
+        :type user_agent: str
+        :param retry_total: The total number of attempts to redo the failed operation when an error happened. Default
+         value is 3.
+        :type retry_total: int
+        :param transport_type: The type of transport protocol that will be used for communicating with
+         the Event Hubs service. Default is ~azure.eventhub.TransportType.Amqp.
+        :type transport_type: ~azure.eventhub.TransportType
+        :param partition_manager: stores the load balancing data and checkpoint data when receiving events
+         if partition_manager is specified. If it's None, this EventHubConsumerClient instance will receive
+         events without load balancing and checkpointing.
+        :type partition_manager: PartitionManager.
+        :param load_balancing_interval: When load balancing kicks in, this is the interval in seconds
+         between two load balancing. Default 10.
+        :type load_balancing_interval: float
+        """
+
+        super(EventHubConsumerClient, self).__init__(
+            host=host, event_hub_path=event_hub_path, credential=credential, **kwargs)
+        self._partition_manager = kwargs.get("partition_manager")
```

Comment:
To bounce back to what we said before, looking at the hierarchy parent classes don't care about `partition_manager`, and there is no need to have it in kwargs (it seems).

Reply:
How about popping partition_manager first and then call parent.

```python
self._partition_manager = kwargs.pop("partition_manager")
self._load_balancing_interval = kwargs.pop("load_balancing_interval", 10)
super(EventHubConsumerClient, self).__init__(..., **kwargs)
```